pulp_rpm/app/downloaders.py

```diff
@@ -196,6 +197,34 @@ async def _run(self, extra_data=None):
             client.close()
         return to_return
 
+    async def _login_and_retry(self, client, max_attempts=5, delay=1):
+        """
+        Attempts to log in using the provided client. Retries up to `max_attempts`
+        times with a delay between attempts.
+
+        Args:
+            client: The client object used for authentication.
+            max_attempts (int): Maximum number of login attempts.
+            delay (int or float): Delay in seconds between retries.
+
+        Returns:
+            The session key from the login.
+
+        Raises:
+            UlnCredentialsError: If all login attempts fail.
+        """
+        for attempt in range(1, max_attempts + 1):
```

Comment:
FWIW we do have the `backoff` crate for this which we already use in pulpcore: https://github.com/pulp/pulpcore/blob/139955c975af0c4bbaad2096f25808206b4d67a3/pulpcore/download/http.py#L250

With that said, this code isn't complex, I'm unsure whether we actually care to pull in a new dependency to the RPM plugin that may conflict with pulpcore at some point if we're not careful

Does the connection seem to drop specifically on the login attempt, but not during the downloads themselves?  Or I suppose presumably the existing retry logic might be handling that.

Reply:
@ggainey Do you have an opinion.  I'm a bit partial to leaving it as-is

Reply:
> Does the connection seem to drop specifically on the login attempt, but not during the downloads themselves? Or I suppose presumably the existing retry logic might be handling that.

Yes the connection drops on the login attempt. This happens a lot during a sync and will throw an error. Adding the login retry fixes the issue.

> Also I would prefer to keep the default number of retries consistent with RpmRemote, so 4.

I can change this.

Reply:
@hstct Thanks, otherwise this PR looks fine